lib/spack/spack/cmd/list.py

```diff
@@ -39,26 +41,45 @@ def setup_parser(subparser):
     subparser.add_argument(
         '-i', '--insensitive', action='store_true', default=False,
         help='Filtering will be case insensitive.')
+    subparser.add_argument(
+        '-s', '--search_description', action='store_true', default=False,
```

Comment:
Also the long argument should be `--search-description`, not `--search_description`